### ISSUE TITLE
Segfault if missing udisks (Qt4)

### ISSUE THREAD

**Issue by 0pLuS0** [rodlie/qtfm#28]
Hi,

First let me say this is awesome to see qtfm back in development! 

```My Specs```
Slackware 14.2 x86_64
Openbox 3.6.1
file-5.25
qt-4.8.7
xorg-server-1.18.3
nvidia-driver-396.24
upower-0.9.23

```I don't have udisks installed...```

I compiled 6.0.0 from source using these cmds in a slackware build script;

QMAKE_CFLAGS="$SLKCFLAGS"
QMAKE_CXXFLAGS="$SLKCFLAGS"

mkdir -p build
cd build
  qmake CONFIG+=debug PREFIX=/usr ..
  make
  make install INSTALL_ROOT=$PKG

When I run qtfm from the term, it just seg faults, this is all I see;

~ >qtfm 
zsh: segmentation fault  qtfm

Below is output from gdb....

Reading symbols from qtfm...done.
(gdb) r
Starting program: /usr/bin/qtfm 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".

Program received signal SIGSEGV, Segmentation fault.
0x00007ffff66b15c0 in QVariant::QVariant(QVariant const&) () from /usr/lib64/qt/lib/libQtCore.so.4
(gdb) bt full
#0  0x00007ffff66b15c0 in QVariant::QVariant(QVariant const&) () at /usr/lib64/qt/lib/libQtCore.so.4
#1  0x00007ffff7996de8 in QDBusPendingReplyData::argumentAt(int) const () at /usr/lib64/qt/lib/libQtDBus.so.4
#2  0x0000000000489e4f in QDBusPendingReply<QString, void, void, void, void, void, void, void>::argumentAt<0>() const (this=0x7fffffffd778)
    at /usr/lib64/qt/include/QtDBus/qdbuspendingreply.h:176
#3  0x00000000004890b7 in QDBusPendingReply<QString, void, void, void, void, void, void, void>::value() const (this=0x7fffffffd778)
    at /usr/lib64/qt/include/QtDBus/qdbuspendingreply.h:181
#4  0x00000000004881e6 in uDisks2::getDevices() () at ../../libdisks/udisks2.h:161
        result = 
              {<QList<QString>> = {{p = {static shared_null = {ref = {_q_value = 97}, alloc = 0, begin = 0, end = 0, sharable = 1, array = {0x0}}, d = 0x6c63a0 <QListData::shared_null>}, d = 0x6c63a0 <QListData::shared_null>}}, <No data fields>}
        call = {d_ptr = 0x8f15c0}
        reply = {<QDBusPendingReplyData> = {<QDBusPendingCall> = {d = {d = 0x8f2950}}, <No data fields>}, <No data fields>}
        devices = 
            {{p = {static shared_null = {ref = {_q_value = 97}, alloc = 0, begin = 0, end = 0, sharable = 1, array = {0x0}}, d = 0x6c63a0 <QListData::shared_null>}, d = 0x6c63a0 <QListData::shared_null>}}
        xml = {d_ptr = {d = 0x7ffff799b4c0}}
#5  0x0000000000483e45 in Disks::scanDevices() (this=0x8e65a0) at ../../libdisks/disks.cpp:129
        foundDevices = 
              {<QList<QString>> = {{p = {static shared_null = {ref = {_q_value = 97}, alloc = 0, begin = 0, end = 0, sharable = 1, array = {0x0}}, d = 0x6c63a0 <QListData::shared_null>}, d = 0x6c63a0 <QListData::shared_null>}}, <No data fields>}
#6  0x0000000000483d24 in Disks::setupDBus() (this=0x8e65a0) at ../../libdisks/disks.cpp:123
        system = {static staticMetaObject = {d = {superdata = 0x0, stringdata = 0x7ffff799e8e0 "QDBusConnection", data = 0x7ffff799eb00, extradata = 0x0}}, d = 0x76bff0}
#7  0x0000000000483a03 in Disks::Disks(QObject*) (this=0x8e65a0, parent=0x0) at ../../libdisks/disks.cpp:110
#8  0x0000000000419cb4 in MainWindow::MainWindow() (this=0x7fffffffde20) at ../../fm/src/mainwindow.cpp:50
        temp = 
          {static null = {<No data fields>}, static shared_null = {ref = {_q_value = 117}, alloc = 0, size = 0, data = 0x6c60fa <QString::shared_null+26>, clean = 0, simpletext = 0, righttoleft = 0, asciiCache = 0, capacity = 0, reserved = 0, array = {0}}, static shared_empty = {ref = {_q_value = 2}, alloc = 0, size = 0, data = 0x7ffff69f70da <QString::shared_empty+26>, clean = 0, simpletext = 0, righttoleft = 0, asciiCache = 0, capacity = 0, reserved = 0, array = {0}}, d = 0x7ffff6504568, static codecForCStrings = 0x0}
        tmp = 
          {static null = {<No data fields>}, static shared_null = {ref = {_q_value = 117}, alloc = 0, size = 0, data = 0x6c60fa <QString::shared_null+26>, clean = 0, simpletext = 0, righttoleft = 0, asciiCache = 0, capacity = 0, reserved = 0, array = {0}}, static shared_empty = {ref = {_q_value = 2}, alloc = 0, size = 0, data = 0x7ffff69f70da <QString::shared_empty+26>, clean = 0, simpletext = 0, righttoleft = 0, asciiCache = 0, capacity = 0, reserved = 0, array = {0}}, d = 0x92f, static codecForCStrings = 0x0}
        realMime = 255
        hl1 = 0x7fffffffe178
        page2 = 0xe
        args = 
              {<QList<QString>> = {{p = {static shared_null = {ref = {_q_value = 97}, alloc = 0, begin = 0, end = 0, sharable = 1, array = {0x0}}, d = 0x7ffff651d948}, d = 0x7ffff651d948}}, <No data fields>}
        name = 
          {static null = {<No data fields>}, static shared_null = {ref = {_q_value = 117}, alloc = 0, size = 0, data = 0x6c60fa <QString::shared_null+26>, clean = 0, simpletext = 0, righttoleft = 0, asciiCache = 0, capacity = 0, reserved = 0, array = {0}}, static shared_empty = {ref = {_q_value = 2}, alloc = 0, size = 0, data = 0x7ffff69f70da <QString::shared_empty+26>, clean = 0, simpletext = 0, righttoleft = 0, asciiCache = 0, capacity = 0, reserved = 0, array = {0}}, d = 0x92f, static codecForCStrings = 0x0}
---Type <return> to continue, or q <return> to quit---

**Comment by rodlie** [comment]
Hi,

Thanks for reporting issues. First off, I develop on Slackware 14.2 in Openbox, so it's a bit odd to see you are having issues.

Try this:
```
git clone https://github.com/rodlie/qtfm
cd qtfm
git checkout 6.0.0
git submodule update -i (this step has been removed in 6.1)
mkdir build && cd build
qmake CONFIG+=release ..
make -j4
./fm/qtfm
```

Btw, I have told the maintainer of qtfm in SlackBuilds to update, but I think he is on vacation or something.

**Comment by rodlie** [comment]
A diff for qtfm 5.5 Slackbuild:
[qtfm-SBo-5.5_to_6.0.0.diff.txt](https://github.com/rodlie/qtfm/files/2172452/qtfm-SBo-5.5_to_6.0.0.diff.txt)

**Comment by rodlie** [comment]
Bug confirmed (I removed udisks in Slackware), will fix as soon as possible.

This is on me, I didn't test QtFM without udisks.

**Comment by rodlie** [comment]
Issue should be fixed in master, will add to the 6.0 branch and do a 6.0.1 release this weekend. 

The reason I didn't catch this is that I develop using Qt5(.9), and this issue only happened when using Qt4.

**Comment by rodlie** [comment]
I have added the patch to the 6.0 branch, you can test with:
```
git clone https://github.com/rodlie/qtfm
cd qtfm
git checkout 6.0
git submodule update -i
mkdir build && cd build
qmake ..
make -j4
./fm/qtfm
```

**Comment by rodlie** [comment]
Ah, I though this (bookmark icons) was just an issue in master, will fix. 

I take request :) Just add them in the issue tracker.

Note that 6.1 is closed for major changes.

Multi panes will need some work, the current code-base was not made for multiple instances, even the tab feature is "fake". I will need to re-structure a lot of stuff before this can happen, probably a 6.3 feature.

**Comment by rodlie** [comment]
Update on the bookmark icons. This is yet again a Qt4 issue, works as it should on Qt5. I will try to fix this, but sooner or later I will depreciate Qt4. The only reason I still maintain Qt4 compatibility is because Slackware still uses it.

**Comment by rodlie** [comment]
The bookmark issue has been fixed, just update the 6.0 branch and rebuild.

I recommend you delete the config dir before testing (~/.config/qtfm6).

**Comment by 0pLuS0** [comment]
Hi,

Well hopefully new Slack release coming, it's just hit two years, so next will be QT5 in Slack.

Ok, I'll make a request post for a few things.

Keep up the great work, I really believe we need a nicer lighter file manager for Linux, even SpaceFM has to many things going on...
